src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -135,13 +141,48 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
```

Comment:
Please explain what this function produces for different episode format.

Reply:
Usually this function will join all the matching groups of the above regular expression with the 'x' character. This means for a regular expression that matches S01E02, it will produce the string '1x2'.